qutip/bloch.py

```diff
@@ -394,6 +403,97 @@ def add_annotation(self, state_or_vector, text, **kwargs):
                                  'text': text,
                                  'opts': kwargs})
 
+    def add_arc(self, start, end, fmt = "b", steps = None, **kwargs):
+        """Adds an arc between two points on a sphere. The arc is set to be
+        blue solid curve by default.
+        
+        Parameters
+        ----------
+        point1 : array_like
+            Array with cartesian coordinates of the first point.
+        point2 : array_like
+            Array with cartesian coordinates of the second point.
+        """
+        if isinstance(start, Qobj):
+            start = [start]
+            for st in start:
+                pt1 = [expect(sigmax(), st),
+                           expect(sigmay(), st),
+                           expect(sigmaz(), st)]
+        else:
+            pt1 = start
+
+        if isinstance(end, Qobj):
+            end = [end]
+            for en in end:
+                pt2 = [expect(sigmax(), en),
+                          expect(sigmay(), en),
+                          expect(sigmaz(), en)]
+        else:
+            pt2 = end
+        pt1 = np.asarray(pt1)
+        pt2 = np.asarray(pt2)
+        
+        len1 = np.linalg.norm(pt1)
+        len2 = np.linalg.norm(pt2)
+        if len1 < 1e-12 or len2 < 1e-12:
+            raise ValueError('Polar and azimuthal angles undefined at origin.')
+        elif abs(len1 - len2) > 1e-12:
+            raise Exception("Points not on the same sphere.")
```

Comment:
```suggestion
            raise ValueError("Points not on the same sphere.")
```

Reply:
When possible, it is best to avoid raising the general `Exception` and instead raise a particular subclass of exception. In this case `ValueError` fits best.